cloudsql/sqlserver/database-sql/cloudsql.go

```diff
@@ -0,0 +1,306 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Sample database-sql demonstrates connection to a Cloud SQL instance from App Engine
+// standard. The application is a Golang version of the "Tabs vs Spaces" web
+// app presented at Cloud Next '19 as seen in this video:
+// https://www.youtube.com/watch?v=qVgzP3PsXFw&t=1833s
+package main
+
+import (
+	"database/sql"
+	"fmt"
+	"html/template"
+	"log"
+	"math"
+	"net/http"
+	"os"
+	"strconv"
+
+	_ "github.com/denisenkom/go-mssqldb"
+)
+
+// parsedTemplate is the global parsed HTML template.
+var parsedTemplate *template.Template
+
+// vote struct contains a single row from the votes table in the database.
+// Each vote includes a candidate ("TABS" or "SPACES") and a timestamp.
+type vote struct {
+	Candidate string
+	VoteTime  sql.NullTime
+}
+
+// voteDiff is used to provide a string representation of the current voting
+// margin, such as "1 vote" (singular) or "2 votes" (plural).
+type voteDiff int
+
+func (v voteDiff) String() string {
+	if v == 1 {
+		return "1 vote"
+	}
+	return strconv.Itoa(int(v)) + " votes"
+}
+
+// templateData struct is used to pass data to the HTML template.
+type templateData struct {
+	TabsCount   uint
+	SpacesCount uint
+	VoteMargin  string
+	RecentVotes []vote
+}
+
+// App struct contains global state.
+type App struct {
+	// db is the global database connection pool.
+	db *sql.DB
+}
+
+// indexHandler handles requests to the / route.
+func (app *App) indexHandler(w http.ResponseWriter, r *http.Request) {
+	switch r.Method {
+	case "GET":
+		if err := showTotals(w, r, app); err != nil {
+			log.Printf("showTotals: %v", err)
+			http.Error(w, "Internal Server Error", http.StatusInternalServerError)
+		}
+	case "POST":
+		if err := saveVote(w, r, app); err != nil {
+			log.Printf("saveVote: %v", err)
+			http.Error(w, "Internal Server Error", http.StatusInternalServerError)
+		}
+	default:
+		http.Error(w, fmt.Sprintf("HTTP Method %s Not Allowed", r.Method), http.StatusMethodNotAllowed)
+	}
+}
+
+func main() {
+
+	var err error
+
+	parsedTemplate, err = template.ParseFiles("templates/index.html")
+	if err != nil {
+		log.Fatalf("unable to parse template file: %s", err)
+	}
+
+	app := &App{}
```

Comment:
This will change to something like:
```suggestion
	app := &App{
		tmpl: tmpl,	
	}
```